packages/b/bullet3/xmake.lua

```diff
@@ -3,19 +3,55 @@ package("bullet3")
     set_description("Bullet Physics SDK.")
 
-    set_urls("https://github.com/bulletphysics/bullet3/archive/$(version).zip",
-             "https://github.com/bulletphysics/bullet3.git")
-    add_versions("2.88", "f361d10961021a186b80821cfc1cfafc8dac48ce35f7d5e8de0943af4b3ddce4")
+    add_urls("https://github.com/bulletphysics/bullet3/archive/$(version).zip", {alias = "zip"})
+    add_urls("https://github.com/bulletphysics/bullet3.git", {alias = "git"})
+    add_versions("zip:2.88", "f361d10961021a186b80821cfc1cfafc8dac48ce35f7d5e8de0943af4b3ddce4")
+    add_versions("git:2.88", "126b6762200b1a9ec8df39b8fd53cb2ad62fb146")
+    add_versions("zip:3.05", "e7ef322d8038e397cd6d79145a856cf5b4d558ce091d49b5239d625a46fef0d7")
+    add_versions("git:3.05", "b12da620abdff665d53cd953118a741758f28523")
+
+    add_configs("double_precision", { description = "Enable double precision floats", default = false, type = "boolean"})
+    add_configs("extras",           { description = "Build the extras", default = false, type = "boolean"})
 
     add_deps("cmake")
 
-    on_install("macosx", "linux", function (package)
-        local configs = {"-DBUILD_CPU_DEMOS=OFF"}
+    on_load("macosx", "linux", "windows", function (package)
+        package:add("linkdirs", package:installdir("lib"))
+    end)
+
+    on_install("macosx", "linux", "windows", function (package)
+        local configs = {"-DBUILD_CPU_DEMOS=OFF", "-DBUILD_OPENGL3_DEMOS=OFF", "-DBUILD_BULLET2_DEMOS=OFF", "-DBUILD_UNIT_TESTS=OFF"}
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DUSE_DOUBLE_PRECISION=" .. (package:config("double_precision") and "ON" or "OFF"))
+        table.insert(configs, "-DBUILD_EXTRAS=" .. (package:config("extras") and "ON" or "OFF"))
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
         import("package.tools.cmake").install(package, configs)
+
+        os.cp("src/*", package:installdir("include", "bullet"))
+        os.rm(path.join(package:installdir("include", "bullet"), "**.cpp"))
+        os.rm(path.join(package:installdir("include", "bullet"), "**.c"))
+        os.rm(path.join(package:installdir("include", "bullet"), "**.txt"))
+        os.rm(path.join(package:installdir("include", "bullet"), "**.lua"))
+        os.rm(path.join(package:installdir("include", "bullet"), "**.cl"))
+
+        os.trycp("**.lib", package:installdir("lib"))
+        os.trycp("**.a", package:installdir("lib"))
+        os.trycp("**.dll", package:installdir("lib"))
+
+        package:add("linkdirs", package:installdir("lib"))
+        package:add("links", "Bullet2FileLoader", "Bullet3Collision", "Bullet3Common", "Bullet3Dynamics", "Bullet3Geometry", "Bullet3OpenCL_clew", "BulletCollision", "BulletDynamics", "BulletInverseDynamics", "BulletSoftBody", "LinearMath")
+        package:add("includedirs", "include", path.join("include", "bullet"))
```

Comment:
Why? Does the default `include` directory not work? 

Even if you want to add, please use `add_inlcudedirs("include", "include/bullet")`

Reply:
When I switched back to ```add_includedirs```, there was no ```include``` directory before I added it manually.